### PR TITLE
Remove choices as values in 3.0

### PR DESCRIPTION
**Pull request by weaverryan**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 3.0+ |
| Fixed tickets | n/a |

I think we must have just forgotten to remove this option!

### CODE REVIEW

**Comment by WouterJ** [comment]
Well, it still exists (otherwise it would be a BC break). Only the default value changed in 3.0. It is deprecated as of Symfony 3.1.

File: reference/forms/types/choice.rst [hunk 1]
```diff
@@ -19,7 +19,6 @@ To use this field, you must specify *either* ``choices`` or ``choice_loader`` op
 |             | - `choice_name`_                                                             |
 |             | - `choice_translation_domain`_                                               |
 |             | - `choice_value`_                                                            |
-|             | - `choices_as_values`_                                                       |
```
Comment (HeahDude) [inline reference/forms/types/choice.rst]: It is deprecated as of 3.1, not removed :)

File: reference/forms/types/choice.rst [hunk 2]
```diff
@@ -192,47 +189,6 @@ would replace the ``choices`` option.
...
-
-* Since 2.7::
-
-    $builder->add('gender', ChoiceType::class, array(
-        // Shows "Male" to the user, returns "m" when selected
-        'choices' => array('Male' => 'm', 'Female' => 'f'),
-        'choices_as_values' => true,
-    ));
-
-In Symfony 3.0, the ``choices_as_values`` option doesn't exist, but the ``choice``
```
Comment (HeahDude) [inline reference/forms/types/choice.rst]: "does not exist" => "is not used"

**Comment by xabbuh** [comment]
Technically it is not deprecated in 3.0, but I agree with Ryan to remove it from the as it does not support setting it to anything else than `true` and will trigger an exception otherwise.

**Comment by HeahDude** [comment]
closes #6474.

@xabbuh Shouldn't we just say that it's not used anymore and that setting `false` will throw an exception then mark it as deprecated in 3.1+ doc ?

**Comment by xabbuh** [comment]
@HeahDude IIRC we always removed the description of features in the version they got deprecated (but here it doesn't make sense to document it imo).

**Comment by HeahDude** [comment]
There were many issues on the repo and and questions on stack, although it is recommended to upgrade to 2.8 before upgrading to 3.0, not everyone does that 😢. I vote (without hard feeling) to keep it documented at the strict minimum by removing the "before/after" stuff but keeping a notice for the legacy reason of the default value. What do you think ?

**Comment by weaverryan** [comment]
Now that I've read the discussion here, I think we should keep the option, but change the description to describe that it's not used anymore, and was something that helped you upgrade. Then, if people see this option in their code and look in the docs for it, they'll see the description.

**Comment by xabbuh** [comment]
Okay, then let's keep the option and update the description instead.

**Comment by HeahDude** [comment]
Nice! Thanks!

**Comment by HeahDude** [comment]
@weaverryan any idea why we the stars is "needs work" here ?

**Comment by WouterJ** [comment]
@HeahDude because we didn't update the label to "finished" before merging.

**Pull request merged**